agent/job_runner.go

```diff
@@ -498,7 +525,14 @@ func (r *JobRunner) createEnvironment() ([]string, error) {
 
 	// We know the BUILDKITE_BIN_PATH dir, because it's the path to the
 	// currently running file (there is only 1 binary)
-	dir, _ := filepath.Abs(filepath.Dir(os.Args[0]))
+	exePath, err := os.Executable()
+	if err != nil {
+		return nil, err
+	}
+	dir, err := filepath.Abs(filepath.Dir(exePath))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nice find. I assume this is a clean up similar to #1444? Is it required for this PR, or just a thing you noticed while changing this part of the code?

Reply:
Just a thing I noticed while here and I debated about splitting it out into is own change set.

This one in particular could introduce a new error case. Previously `BUILDKITE_BIN_PATH` may have been set to an empty string but the function succeeded, now it would error. What do you think?

Reply:
I don't have a strong view about splitting it out or not, but I lean towards splitting it only because I assume the review intensity and feedback cycle count will be very different for adding the new hook vs. improving the binary path generation. I assume the latter will land very easily and Just Be Done.

> This one in particular could introduce a new error case. Previously BUILDKITE_BIN_PATH may have been set to an empty string but the function succeeded, now it would error. What do you think?

I assume the empty string case would be rare? If so, this seems like a behaviour change that we can accept?